client/src/components/AllButtons.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from "react";
+//import videosData from "../exampleresponse.json";
+import Card from "./Card";
+import Form from "./Form";
+
+const Api = "https://junita-full-stack-project-assessment.onrender.com/";
+
+function AllButtons() {
+  const [videos, setVideos] = useState([]);
+
+  useEffect(() => {
+    fetch(Api)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("Fetched Data:", data);
+        const videosWithRating = data.map((video) => ({
+          ...video,
+          rating: video.rating || 0,
```

Comment:
Why do you need to do this? What situation leads to you getting a video with no rating?